internal/provider/auth.go

```diff
@@ -270,11 +271,44 @@ func (l *AuthLoginCommon) init(d *schema.ResourceData) (string, map[string]inter
 	return path, params, nil
 }
 
-func (l *AuthLoginCommon) checkRequiredFields(d *schema.ResourceData, required ...string) error {
+type authDefault struct {
+	field string
+
+	// envVars will override defaultVal
+	envVars []string
+	// defaultVal is the fallback if an env var is not set
+	defaultVal string
+}
+
+type authDefaults []authDefault
+
+func (l *AuthLoginCommon) setDefaultFields(d *schema.ResourceData, defaults authDefaults, params map[string]interface{}) error {
+	for _, f := range defaults {
+		if _, ok := l.getOk(d, f.field); !ok {
+			// if field is unset in the config, check env
+			for _, envVar := range f.envVars {
+				defaultValue := os.Getenv(envVar)
```

Comment:
nit: variable names are bit odd, it seems like the  only true `defaultValue` is `f.defaultVal` - `envVal` might make it a bit easier to read.